build-tools/build-package

```diff
@@ -0,0 +1,527 @@
+// Delete the existing package.json so that if this process fails it makes
+// it more difficult to accidentally publish the corrupted state
+if (fs.existsSync(path.resolve(DIR, 'package.json'))) {
+	fs.unlinkSync(path.resolve(DIR, 'package.json'));
+}
+
+const DATA = path.resolve(DIR, 'data');
+if (!fs.existsSync(DATA)) fs.mkdirSync(DATA);
+
+// Due to circular dependencies caused by the nature of how PS is architected around
+// globals the sanest way to create a usable package is to concatenate all of the
+// logic together. The data files are bulkier and would benefit from being downloaded
+// in parallel (or elided completely), so we break these out into their own data files
+// to let the consumer decide how/if they should be included.
+const cpdata = (name, type) => {
+	const o = `Battle${type}`;
+	const t = require(`../data/${Array.isArray(name) ? name[0] : name}`, 'utf8')[o];
+	fs.writeFileSync(path.resolve(DATA, `${Array.isArray(name) ? name[1] : name}.json`), JSON.stringify(t));
+};
+
+cpdata('abilities', 'Abilities');
+cpdata('aliases', 'Aliases');
+cpdata('items', 'Items');
+cpdata('moves', 'Movedex');
+cpdata(['pokedex', 'species'], 'Pokedex');
+cpdata('text', 'Text');
+cpdata(['typechart', 'types'], 'TypeChart');
+
+// BattleTeambuilderTable gets special treatment to remove the fields only relevant to PS's
+// teambuilder internals and preserve the more general fields required for ModdedDex to work
+const BattleTeambuilderTable = require('../data/teambuilder-tables.js').BattleTeambuilderTable;
+
+const table = {};
+// The 'lite' learnsets included in the BattleTeambuilderTable are not required for regular
+// operation but are collected separately to write to a 'learnsets.json' file that is not
+// part of the default exports but may be loaded on demand
+const learnsets = {};
+const SKIP = new Set(['tiers', 'items', 'formatSlices']);
+for (const m in BattleTeambuilderTable) {
+	if (SKIP.has(m)) continue;
+	if (m === 'learnsets') {
+		learnsets[m] = BattleTeambuilderTable[m];
+		continue;
+	}
+	const mod = {};
+	const ls = {};
+	for (const f in BattleTeambuilderTable[m]) {
+		if (SKIP.has(f)) continue;
+		if (f === 'learnsets') {
+			ls[f] = BattleTeambuilderTable[m][f];
+			continue;
+		}
+		mod[f] = BattleTeambuilderTable[m][f];
+	}
+	if (Object.keys(mod).length) table[m] = mod;
+	if (Object.keys(ls).length) learnsets[m] = ls;
+}
+
+fs.writeFileSync(path.resolve(DATA, 'formats-data.json'), JSON.stringify(table));
+fs.writeFileSync(path.resolve(DATA, 'learnsets.json'), JSON.stringify(learnsets));
+
+// TypeScript assumes each individual entry is its own type and doesn't attempt
+// to perform unification to end up with a more general type. As such, the JSON
+// data is cast to slightly simplified versions of the types lifted from the
+// server code in our main bundle. When then set up each of the constants on
+// window where some of the code looks for it.
+let index =
+`import * as abilitiesJSON from './data/abilities.json';
```

Comment:
This is a huge string; shouldn't it just get its own file?

Reply:
I thought about that, but its not really a valid ts file on its own. I dont really have a strong preference though, i liked it here because then everything is contained in one place and we dont add a 'package-prelude.ts` which we also add to the tsconfig exclude and have to read in here instead.